Support tsconfig.json

wassertim opened the pull request:
This is still work in progress. The idea is to drop the settings via sbt keys and use tsconfig instead.

Any remarks and comments are appreciated.

I think we should use only tsconfig for the settings, so this is a breaking change.

- [x] Fix the tests (introduce tsconfig.json in all tests)
- [x] Fix double compilation problem
- [x] update readme to reflect the changes
- [x] fix the wrong line number on errors 

Not all the settings of tsconfig could be used (outDir, include, exclude) due to the sbt-plugin specifics.

wassertim commented on src/main/resources/typescriptc.js:
```diff
@@ -72,9 +74,17 @@
         logger.debug("creating compiler settings");
-        var compSettings = new typescript.getDefaultCompilerOptions();
+
+        var config = getTSConfig(options);
```
that is where the tsconfig is being parsed

BrandonArp commented on build.sbt:
```diff
@@ -24,11 +24,13 @@ organization := "com.arpnetworking"
 
 name := "sbt-typescript"
 
+version := "0.2.3.10-SNAPSHOT"
```
This should be in version.sbt

wassertim replied:
cool, thanks, I was wondering where it is gone :)

wassertim commented:
@BrandonArp I'm done with tsconfig support but I'm still refactoring it (because, apparently no one has done it since the beginning). I added some es6 features and cleaned the compile method, moving out the functions. You can see the progress with every commit. Please let me know if you have some remarks and suggestions.

BrandonArp commented:
The refactoring is looking great!  Let me know when you're good with me merging it.

We'll need to update the README file to reflect the new usage.  We should also probably bump the Y version to 0.3.0-SNAPSHOT since this is a breaking change (and I don't quite feel comfortable calling it 1.0 yet).

wassertim commented:
Yes, I will update the README file as well. I will let you know when it is done, so you will take a look again.

wassertim commented:
@BrandonArp could you please review it again. It is ready to be merged.

wassertim commented:
@BrandonArp Hey, I think we can make the world a little bit better if we merge it :) Everything's done, including the readme

BrandonArp commented:
Thank you for all the hard work.  This looks great.
